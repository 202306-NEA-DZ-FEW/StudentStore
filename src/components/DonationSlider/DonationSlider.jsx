import Image from "next/image";
import SwiperCore, { Autoplay } from "swiper/core";
import { Swiper, SwiperSlide } from "swiper/react";

import "swiper/swiper-bundle.min.css";

const DonationSlider = () => {
    SwiperCore.use([Autoplay]);
    return (
        <div className='carousel w-[90%] md:w-[70%] mx-auto relative'>
            <Swiper
                slidesPerView={1}
                autoplay={{
                    delay: 2500,
                }}
                loop={true}
            >
                <SwiperSlide>
                    <div
                        id='slide1'
                        className='carousel-item relative h-[70%] w-full'
                    >
                        <Image
                            src='/images/donation1.jpg'
                            className='mx-auto '
                            layout='responsive'
                            width={700}
                            height={700}
                            alt='donation image'
                        />
                    </div>
                </SwiperSlide>
                <SwiperSlide>
                    <div
                        id='slide2'
                        className='carousel-item relative h-[70%] w-full'
                    >
                        <Image
                            src='/images/donation2.jpg'
                            className='mx-auto'
                            width={700}
                            height={700}
                            layout='responsive'
                            alt='donation image'
                        />
                    </div>
                </SwiperSlide>
                <SwiperSlide>
                    <div
                        id='slide3'
                        className='carousel-item relative h-[70%] w-full'
                    >
                        <Image
                            src='/images/donation3.jpg'
                            className='mx-auto'
                            width={700}
                            height={700}
                            layout='responsive'
                            alt='donation image'
                        />
                    </div>
                </SwiperSlide>
                ...
            </Swiper>
        </div>
    );
};

export default DonationSlider;
