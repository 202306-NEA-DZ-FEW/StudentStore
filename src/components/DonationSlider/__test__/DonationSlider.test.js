import renderer from "react-test-renderer";

import DonationSlider from "../DonationSlider";

it("renders correctly", () => {
    const tree = renderer.create(<DonationSlider />).toJSON();
    expect(tree).toMatchSnapshot();
});
