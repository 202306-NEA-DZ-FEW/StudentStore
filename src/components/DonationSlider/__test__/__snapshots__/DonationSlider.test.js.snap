// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="carousel w-[90%] md:w-[70%] mx-auto relative"
>
  <div
    className="swiper-container"
  >
    <div
      className="swiper-wrapper"
    >
      <div
        className="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index={2}
      >
        <div
          className="carousel-item relative h-[70%] w-full"
          id="slide3"
        >
          <img
            alt="donation image"
            className="mx-auto"
            data-nimg="1"
            decoding="async"
            height={700}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            sizes="100vw"
            src="/_next/image?url=%2Fimages%2Fdonation3.jpg&w=3840&q=75"
            srcSet="/_next/image?url=%2Fimages%2Fdonation3.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fdonation3.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fdonation3.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fdonation3.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fdonation3.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fdonation3.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fdonation3.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fdonation3.jpg&w=3840&q=75 3840w"
            style={
              Object {
                "color": "transparent",
                "height": "auto",
                "width": "100%",
              }
            }
            width={700}
          />
        </div>
      </div>
      <div
        className="swiper-slide"
        data-swiper-slide-index={0}
      >
        <div
          className="carousel-item relative h-[70%] w-full"
          id="slide1"
        >
          <img
            alt="donation image"
            className="mx-auto "
            data-nimg="1"
            decoding="async"
            height={700}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            sizes="100vw"
            src="/_next/image?url=%2Fimages%2Fdonation1.jpg&w=3840&q=75"
            srcSet="/_next/image?url=%2Fimages%2Fdonation1.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fdonation1.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fdonation1.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fdonation1.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fdonation1.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fdonation1.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fdonation1.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fdonation1.jpg&w=3840&q=75 3840w"
            style={
              Object {
                "color": "transparent",
                "height": "auto",
                "width": "100%",
              }
            }
            width={700}
          />
        </div>
      </div>
      <div
        className="swiper-slide"
        data-swiper-slide-index={1}
      >
        <div
          className="carousel-item relative h-[70%] w-full"
          id="slide2"
        >
          <img
            alt="donation image"
            className="mx-auto"
            data-nimg="1"
            decoding="async"
            height={700}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            sizes="100vw"
            src="/_next/image?url=%2Fimages%2Fdonation2.jpg&w=3840&q=75"
            srcSet="/_next/image?url=%2Fimages%2Fdonation2.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fdonation2.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fdonation2.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fdonation2.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fdonation2.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fdonation2.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fdonation2.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fdonation2.jpg&w=3840&q=75 3840w"
            style={
              Object {
                "color": "transparent",
                "height": "auto",
                "width": "100%",
              }
            }
            width={700}
          />
        </div>
      </div>
      <div
        className="swiper-slide"
        data-swiper-slide-index={2}
      >
        <div
          className="carousel-item relative h-[70%] w-full"
          id="slide3"
        >
          <img
            alt="donation image"
            className="mx-auto"
            data-nimg="1"
            decoding="async"
            height={700}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            sizes="100vw"
            src="/_next/image?url=%2Fimages%2Fdonation3.jpg&w=3840&q=75"
            srcSet="/_next/image?url=%2Fimages%2Fdonation3.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fdonation3.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fdonation3.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fdonation3.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fdonation3.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fdonation3.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fdonation3.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fdonation3.jpg&w=3840&q=75 3840w"
            style={
              Object {
                "color": "transparent",
                "height": "auto",
                "width": "100%",
              }
            }
            width={700}
          />
        </div>
      </div>
      <div
        className="swiper-slide swiper-slide-duplicate"
        data-swiper-slide-index={0}
      >
        <div
          className="carousel-item relative h-[70%] w-full"
          id="slide1"
        >
          <img
            alt="donation image"
            className="mx-auto "
            data-nimg="1"
            decoding="async"
            height={700}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            sizes="100vw"
            src="/_next/image?url=%2Fimages%2Fdonation1.jpg&w=3840&q=75"
            srcSet="/_next/image?url=%2Fimages%2Fdonation1.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fdonation1.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fdonation1.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fdonation1.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fdonation1.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fdonation1.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fdonation1.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fdonation1.jpg&w=3840&q=75 3840w"
            style={
              Object {
                "color": "transparent",
                "height": "auto",
                "width": "100%",
              }
            }
            width={700}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
